talentsearch/resources/js/components/skills/SkillResults.tsx

```diff
@@ -0,0 +1,129 @@
+  const intl = useIntl();
+  const locale = getLocale(intl);
+  const { id, name, description } = skill;
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <div
+      data-h2-border="b(lightgray, top, solid, s)"
+      data-h2-padding="b(top-bottom, s)"
+    >
+      <div
+        data-h2-display="b(flex)"
+        data-h2-flex-direction="b(row)"
+        data-h2-justify-content="b(space-between)"
+        data-h2-align-items="b(center)"
+      >
+        {isAdded ? (
+          <span
+            data-h2-font-color="b(lightpurple)"
+            data-h2-font-weight="b(700)"
+            data-h2-display="b(flex)"
+            data-h2-align-items="b(flex-start)"
+          >
+            <CheckCircleIcon style={{ width: "1.125rem" }} />
+            <span data-h2-padding="b(left, xxs)">{name[locale]}</span>
+          </span>
+        ) : (
+          <span>{name[locale]}</span>
+        )}
+        <div>
+          <Button
+            color="primary"
+            mode="inline"
+            type="button"
+            onClick={() => setIsOpen(!isOpen)}
+          >
+            {isOpen
+              ? intl.formatMessage({
+                  defaultMessage: "Hide definition",
+                  description: "Text displayed when skill block is open.",
+                })
+              : intl.formatMessage({
+                  defaultMessage: "See definition",
+                  description: "Text displayed when skill block is open.",
+                })}
+          </Button>
+          <Button
+            color="primary"
+            mode="inline"
+            type="button"
+            onClick={
+              isAdded ? () => handleRemoveSkill(id) : () => handleAddSkill(id)
+            }
+          >
+            {isAdded
+              ? intl.formatMessage({
+                  defaultMessage: "Remove skill",
+                  description:
+                    "Button label to remove skill on skill result block.",
+                })
+              : intl.formatMessage({
+                  defaultMessage: "Add skill",
+                  description:
+                    "Button label to add skill on skill result block.",
+                })}
+          </Button>
+        </div>
+      </div>
+      {isOpen && (
+        <div>
+          <p>{description?.[locale]}</p>
+        </div>
+      )}
+    </div>
+  );
+};
+
+export interface SkillResultsProps {
+  title: string;
+  skills: Skill[];
+  addedSkills: Skill[];
+  handleAddSkill: (id: string) => Promise<void>;
+  handleRemoveSkill: (id: string) => Promise<void>;
+}
+
+const SkillResults: React.FunctionComponent<SkillResultsProps> = ({
+  title,
+  skills,
+  addedSkills,
+  handleAddSkill,
+  handleRemoveSkill,
+}) => {
+  return (
+    <section>
+      <h4>
+        {title} ({skills.length})
+      </h4>
+      {skills.map((skill) => {
+        // Check if the poolCandidate has added the skill already.
+        const isAdded = addedSkills.includes(skill);
```

Comment:
This way of checking for added skills only works if the two lists reference the exact same skill objects. Comparing ids would be more reliable.
```suggestion
  addedIds = addedSkills.map((skill) => skill.id);
  return (
    <section>
      <h4>
        {title} ({skills.length})
      </h4>
      {skills.map((skill) => {
        // Check if the poolCandidate has added the skill already.
        const isAdded = addedIds.includes(skill.id);
```

Reply:
Good point, however `Array.includes()` only accepts the value with the same type as the array's values. I'll use the `Array.find()` instead which finds the fist element where the predicate is true (id's match).

Reply:
I'd rather see includes than find, when we just want a boolean "is it in the array" answer. What's wrong with `addedIds.includes(skill.id)`?

Reply:
Oh missed this comment.. ya that works too, we don't need the other data when we just need to check for the matching id's.